src/UI/Component/ViewControl/Pagination.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/* Copyright (c) 2017 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+namespace ILIAS\UI\Component\ViewControl;
+
+use \ILIAS\UI\Component as C;
+use ILIAS\UI\Component\JavaScriptBindable;
+use ILIAS\UI\Component\Triggerer;
+/**
+ * This describes a Pagination Control
+ */
+interface Pagination extends C\Component, JavaScriptBindable, Triggerer {
+
+	/**
+	 * Initialize with the total amount of entries
+	 * of the controlled data-list
+	 *
+	 * @param 	int 	$total
+	 *
+	 * @return \Pagination
+	 */
+	public function withTotalEntries($total);
+
+	/**
+	 * Set the amount of entries per page.
+	 *
+	 * @param 	int 	$size
+	 *
+	 * @return \Pagination
+	 */
+	public function withPageSize($size);
+
+	/**
+	 * Get the numebr of entries per page.
+	 *
+	 * @return int
+	 */
+	public function getPageSize();
+
+	/**
+	 * Set the selected page.
+	 *
+	 * @param 	int 	$page
+	 *
+	 * @return \Pagination
+	 */
+	public function withCurrentPage($page);
+
+	/**
+	 * Get the currently slected page.
+	 *
+	 * @return int
+	 */
+	public function getCurrentPage();
+
+	/**
+	 * Get the data's offset according to current page and page size.
+	 *
+	 * @return int
+	 */
+	public function getOffset();
+
+	/**
+	 * Register a signal with the control.
+	 *
+	 * @param ILIAS\UI\Component\Signal $signal
+	 *
+	 * @return \Pagination
+	 */
+	public function withOnSelect(C\Signal $signal);
+
+	/**
+	 * Calculate the total number of pages.
+	 *
+	 * @return int
+	 */
+	public function getNumberOfPages();
+
+	/**
+	 * Layout; define, how many page-options are shown (max).
+	 *
+	 * @param int 	$amount
+	 *
+	 * @return \Pagination
+	 */
+	public function withMaxPaginationButtons($amount);
```

Comment:
Do you see any chance to control this via some css or client side code instead of setting it here? It seems at if the value one wants to set here depends on some css-induced circumstances (font, paddings, ...) that the developer introducing that element won't know in advance. Another option would be to just fix this and remove this setter. I just cannot see how a developer using this element could sensibly control this.

Reply:
Imo, this is more of a usability decision than a visual thing. I like the option of limiting/unlimiting the pagination, since there might be numbered lists with easy-to-grasp steps, where you explicitly favor quick navigation over cleanliness.
Thus, you would still need a setter. Why not with a value, then?
However, where would you see the fix value? I personally think 5 is too low and 10 is too big, so I would go for 7, if at all.

Reply:
Just leave it as it is.